frontend/src/views/modelers/ModelersView.tsx

```diff
@@ -0,0 +1,107 @@
+import { useMachine } from '@xstate/react';
+import gql from 'graphql-tag';
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { ModelersEmptyView } from './ModelersEmptyView';
+import { ModelersErrorView } from './ModelersErrorView';
+import { ModelersLoadedView } from './ModelersLoadedView';
+import { ModelersLoadingView } from './ModelersLoadingView';
+import {
+  ErrorFetchingEvent,
+  FetchedModelersEvent,
+  ModelersUpdatedEvent,
+  ModelersViewContext,
+  ModelersViewEvent,
+  modelersViewMachine,
+} from './ModelersViewMachine';
+
+const getModelersQuery = gql`
+  query getModelers($projectId: ID!) {
+    viewer {
+      project(projectId: $projectId) {
+        modelers {
+          id
+          name
+          status
+        }
+      }
+    }
+  }
+`;
+
+export const ModelersView = () => {
+  const { projectId } = useParams();
+  const [{ value, context }, dispatch] = useMachine<ModelersViewContext, ModelersViewEvent>(modelersViewMachine);
+  const { modelers, message } = context;
+
+  // Load current project list
+  const { loading: modelersLoading, data: modelersData, error: modelersError } = useQuery(getModelersQuery, {
+    variables: { projectId },
+  });
+  useEffect(() => {
+    if (!modelersLoading) {
+      if (modelersError) {
+        const errorFetching = { type: 'ERROR_FETCHING', message: modelersError.message } as ErrorFetchingEvent;
+        dispatch(errorFetching);
+      } else if (modelersData) {
+        let { modelers } = modelersData?.viewer.project;
+        const fetchedProjects = { type: 'FETCHED_MODELERS', modelers } as FetchedModelersEvent;
+        dispatch(fetchedProjects);
+      }
+    }
+  }, [modelersLoading, modelersData, modelersError, dispatch]);
+
+  // Setup callback to update project list when invoked
+  const [getModelers, { loading, error, data }] = useLazyQuery(getModelersQuery);
+  useEffect(() => {
+    if (!loading) {
+      if (error) {
+        const errorFetching = { type: 'ERROR_FETCHING', message: error.message } as ErrorFetchingEvent;
+        dispatch(errorFetching);
+      } else if (data) {
+        let { modelers } = data?.viewer.project;
+        const modelersUpdatedEvent = { type: 'MODELERS_UPDATED', modelers } as ModelersUpdatedEvent;
+        dispatch(modelersUpdatedEvent);
+      }
+    }
+  }, [loading, error, data, dispatch]);
+
+  let view = null;
+  switch (value) {
+    case 'loading':
+      view = <ModelersLoadingView />;
+      break;
+    case 'empty':
+      view = <ModelersEmptyView />;
+      break;
+    case 'error':
+      view = <ModelersErrorView message={message} />;
+      break;
+    case 'loaded':
+      view = (
+        <ModelersLoadedView
+          modelers={modelers}
+          onModelerUpdated={() => {
+            getModelers({ variables: { projectId } });
+          }}
+        />
+      );
+      break;
+    default:
+      view = <ModelersErrorView message={message} />;
```

Comment:
You could simply separate two situations like that:

if (value === 'empty') {
  return <ModelersEmptyView />;
}

return (
  // The view which will be able to both display some content and open a snackbar to display errors.
);